src/components/organisms/batch-jobs-activity-list/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { bytesConverter } from "../../../utils/bytes-converter"
+import { ActivityCard } from "../../molecules/activity-card"
+import { useBatchJobDescription } from "../../../hooks/use-batch-job-description"
+import FileIcon from "../../fundamentals/icons/file-icon"
+
+const BatchJobActivityList = ({ batchJobs }: { batchJobs?: BatchJob[] }) => {
+  return <div>
+    {
+      !!batchJobs?.length && (
+        batchJobs?.map(batchJob => {
+          return <BatchJobActivityCard key={batchJob.id} batchJob={batchJob}/>
+        })
+      )
+    }
+  </div>
+}
+
+const BatchJobActivityCard = ({ batchJob }: { batchJob: any }) => {
+  const notification = useNotification()
+  const { store } = useAdminStore()
+  const { mutate: cancelBatchJob, error: cancelBatchJobError  } =
+    useAdminCancelBatchJob(batchJob.id)
+
+  const relativeTimeElapsed = getRelativeTime({
+    from: new Date(),
+    to: batchJob.created_at,
+  })
+
+  const batchJobActivityDescription = useBatchJobDescription(
+    batchJob,
+    relativeTimeElapsed.raw
+  )
+
+  const canCancel = batchJob.status !== "completed" &&
+    batchJob.status !== "failed" &&
+    batchJob.status !== "canceled"
+
+  const canDownload = batchJob.status === "completed" &&
+    batchJob.result?.file_key
+
+  useEffect(() => {
+    if (cancelBatchJobError) {
+      notification(
+        "Error canceling the batch job",
+        getErrorMessage(cancelBatchJobError),
+        "error"
+      )
+    }
+  }, [cancelBatchJobError])
+
+  const deleteFile = () => {
+    if (!batchJob.result?.file_key) return
+    Medusa.uploads.delete(batchJob.result?.file_key)
+      .then(() => {
+        notification(
+          "Success",
+          "Export file has been removed",
+          "success"
+        )
+      })
+      .catch(() => {
+        notification(
+          "Error",
+          "Something went wrong while deleting the export file",
+          "error"
+        )
+      })
+  }
+
+  const downloadFile = async () => {
+    if (!batchJob.result?.file_key) return
+    Medusa.uploads.downloadUrl(batchJob.result?.file_key)
+      .then((response) => {
+        const link = document.createElement("a");
+        link.href = response.data.download_url;
+        link.setAttribute(
+            "download",
+            `${batchJob.result?.file_key}`
+        );
+        document.body.appendChild(link);
+        link.click();
+
+        document.body.removeChild(link);
+      })
+      .catch(() => {
+        notification(
+          "Error",
+          "Something went wrong while downloading the export file",
+          "error")
+      })
+  }
+
+  const getBodyAction = () => {
+    return batchJob.result?.file_key && (
+      <DownloadableFileButton
+        onClick={downloadFile}
+        variant={"ghost"}
+        size={"small"}
+        className="inter-small-regular -ml-3 p-2.5 my-1"
+        fileName={batchJob.result.file_key}
+        icon={<FileIcon size={40}/>}
```

Comment:
We should have an action for when the file is loading as well: 
![image](https://user-images.githubusercontent.com/88927411/176187272-0e29e114-02ad-45c6-bdce-b538dcbb4f85.png)

(we have the spinner component already, we should switch on the state of the batch job in order to get the right layout)